pypose/module/dynamics.py

```diff
@@ -0,0 +1,368 @@
+from numpy import vectorize
+import torch as torch
+import torch.nn as nn
+from torch.autograd.functional import jacobian
+
+class _System(nn.Module):
+    r'''
+    A sub-class of :obj:`torch.nn.Module` to build general dynamics.
+    
+    Args:
+        time (:obj:`boolean`): Whether the system is time-varying; defaults to False, meaning time-invariant
+
+    Linearization
+    ----------
+    The nonlinear state-space equation is given as:
+
+    .. math::
+        \begin{align}
+        \overrightarrow{x}_{k+1} &= \mathbf{f}(\overrightarrow{x}_k,\overrightarrow{u}_k) \\
+        \overrightarrow{y}_{k} &= \mathbf{g}(\overrightarrow{x}_k,\overrightarrow{u}_k)
+        \end{align}
+
+    This class provides a means to linearize the system at any point along a trajectory.
+    \bf{Note}: The linearization can be provided for any arbitrary point, not just the equilibrium point(s).
+
+    Suppose we to linearize about :math:`(\overrightarrow{x}^*,\overrightarrow{u}^*)`
+    for an arbitrary equation :math:`h(x,u)`.
+    Through a Taylor series expansion (ignoring higher order terms), we get the following:
+
+    .. math::

```

Comment:
This is somewhat unclear.  I suggest go with the standard perturbation form:
.. math::
    h(x^*+\delta x,u^*+\delta u) = h(x^*,u^*) + \frac{\partial h}{\partial x} \right|_{x^*,u^*} \delta x + blah
Then explain that we would like to work with the full states, so we end up with
.. math::
    h(x^*+\delta x,u^*+\delta u) = \frac{\partial h}{\partial x} \right|_{x^*,u^*} (x^*+\delta x) + blah + c
where then we show the definition of c.